src/Serilog/Settings/KeyValuePairs/KeyValuePairSettings.cs

```diff
@@ -164,12 +161,21 @@ select new
             }
         }
 
+        internal static IReadOnlyDictionary<string, string> ExtractDirectives(IReadOnlyDictionary<string, string> settings)
+        {
+            var directives = settings
+                .Where(kvp => _supportedDirectives.Any(kvp.Key.StartsWith))
+                .ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+
+            return new ReadOnlyDictionary<string, string>(directives);
```

Comment:
Also here, if we don't create a `ReadOnlyDictionary` explicitly, we can get this back to a one-liner and avoid the additional method.

(Nitpick - the codebase doesn't use explicit `internal` modifiers elsewhere.)

Reply:
Actually the method exists for the sake of tests ... but I don't know if the test 

```csharp
+        [Fact]
 +        public void IrrelevantKeysAreIgnored()
 +        {
 +            var irrelevantSettings = new Dictionary<string, string>()
 +            {
 +                ["whatever:foo:bar"] = "willBeIgnored",
 +                ["irrelevant"] = "willBeIgnored",
 +            };
 +
 +            var directives = KeyValuePairSettings.ExtractDirectives(irrelevantSettings);
 +
 +            Assert.False(directives.Any());
 +        }
```

provides much value ...

(about `internal` : dammit, I thought I had removed them all !)

Reply:
Makes sense, I didn't notice the test dependency, thanks 👍